### ISSUE TITLE
Shorter predictions using --nn N on MacOS 

### ISSUE THREAD

**Issue by ivan-pavlov** [JohnLangford/vowpal_wabbit#1441]
I am trying to train sigmoidal feedforward network using `--nn N` argument. 
`vw -d train_data.vw --loss_function logistic -f mdl.vw --nn 10`

But when I make predictions:
`vw -t -i mdl.vw -p preds.vw --link=logistic -d valid_data.vw`
I get shorter prediction file `preds.vw`:
```
wc valid_data.vw preds.vw
   13940  153340 1822570 X_valid.vw
   13907   13907  125331 preds.vw
   27847  167247 1947901 total
```

I have tested this behavior on MacOS X, Win10, and Gentoo Linux. 
So far I can only get such error on MacOS 10.13.1-4 using Vowpal Wabbit version 8.5.0
Other versions are working correctly.

Does anyone have the same problem?

**Comment by JohnLangford** [comment]
This is quite odd.  

Is the length of the shortening always the same?  Are there any other issues observed?  Do you have a reproducible dataset?  If you eliminate some of the flags does the problem persist?

**Comment by ivan-pavlov** [comment]
The length of the shortening is different every time and I haven't observed other issues.

I have added reproducible dataset here:
[valid_data.txt](https://github.com/JohnLangford/vowpal_wabbit/files/1886630/valid_data.txt)
[train_data.txt](https://github.com/JohnLangford/vowpal_wabbit/files/1886632/train_data.txt)

Yes, I have this problematic behavior without any optional flags (except for `--nn N` itself).

**Comment by JohnLangford** [comment]
I'm unable to reproduce this with the current master, so maybe it's fixed?  (I'm running on Ubuntu 16.04.)  

I'm going to close this on the theory that it's fixed, but please reopen if it remains.
